### ISSUE TITLE
Accessing and downloading (xml, pdf ) files from handelsregister API

### ISSUE THREAD

**Issue by timtensor** [bundesAPI/handelsregister#23]
Hi @wirthual , 
I am trying the following workflow 

- Using the advanced search query : https://www.handelsregister.de/rp_web/erweitertesuche.xhtml  i do the following 
              a) Choose Federal States - Berlin / Bavaria 
              b) Company / search words - "Hallo"
              c) Choose the option "contain all words"
              d ) Type of registar "HRA"
              e ) And choose 100 hits per list . 

- This will give us a search list as shown in the screen shot 
- I want to then downlaod SI  or DK content of all the hits . 
I am wondering if that is possible or not ? 
Screenshot : 
![image](https://github.com/bundesAPI/handelsregister/assets/49288450/107f27ae-4672-4150-8982-3f4711b2d270)

Please let me know what could be the best way forward

**Comment by melihsunbul** [comment]
Hello, did you able to figure a way out?

**Comment by timtensor** [comment]
Hi unfortunately not , since it has to come up from the team it is currently kind of blocked

**Comment by melihsunbul** [comment]
> Hi unfortunately not , since it has to come up from the team it is currently kind of blocked

Thanks

**Comment by melihsunbul** [comment]
> interested in other solutions? or do you want to use this one specifically?

I am open to hear other alternatives if there is any.

**Comment by timtensor** [comment]
As I am using a colab notebook , I think an API would be the best right ? I think the colab notebook doesn't support selenium . 

The project basically collects data to analyze different companies .

**Comment by melihsunbul** [comment]
Yeah, I agree. I already tried an alternative to selenium which is playwright, but it has limitations that come together with scraping the website.

**Comment by melihsunbul** [comment]
I already tried to use proxies by rotating, but in that case, the website is not being loaded in a reasonable time. I am planning to send above 200 requests periodically.

**Comment by timtensor** [comment]
Oh ok , I think the API just makes it a bit easier and is there for a reason   , I guess but I think it is not maintained

**Comment by timtensor** [comment]
no sorry actually i have no clue , you managed to get the documents via api ? Like all the documents related to a search . If so it would be great if you could share your method .

**Comment by mauhai** [comment]
I have built a complete solution for this problem and will see how and whether I can share the code and approach somehow. As mentioned above its using full browser rendering as opposed to this API, which I think is rather a dead-end when it comes to actually downloading the documents.

I do hope, that the Handelsregister will at some point publish a proper API.

**Comment by timtensor** [comment]
> I tried and found the solution, I found their API to get the document but that API required a cookie (SESSIONID). Which is not accessible using javascript. Do you have any solution to get the cookie?

Did you find a way to resolve this issue . I was looking into playwright but so far did not manage to find the issue. 
How are you using the API to download the docs ?

**Comment by pmahan00** [comment]
> > download
> 
> For wich project do you want to use the API. If you are working on larg project(downloading millions of documents), i can offer you a paid solutions
Not a large project, but around 15-45 documents to do NLP analysis and run statistics

**Comment by timtensor** [comment]
> > download
> 
> For wich project do you want to use the API. If you are working on larg project(downloading millions of documents), i can offer you a paid solutions 

Small project for about 15-45 documents for NLP analysis

**Comment by m-derra** [comment]
Any news here? I would also need to crawl more detailed information like Unternehmensgegenstand. As I understand it, we need to download the .xml file for that.

**Comment by mauhai** [comment]
@m-derra If you or others still need support here, I built a hosted API with which you can access the respective files. Its not ideal and still in beta, but should solve your problem. I'm hoping to be able to release it to the public in the future. Feel free to reach out to me via hai [at] openregisters [dot] com.
